cpp/src/slice2cpp/Gen.cpp

```diff
@@ -908,20 +950,33 @@ Slice::Gen::generate(const UnitPtr& p)
     {
         normalizeMetaData(p, true);
 
+        // Forward declares proxies, classes and structs.
         DeclVisitor declVisitor(H, C, _dllExport);
         p->visit(&declVisitor, false);
 
-        TypesVisitor typesVisitor(H, C, _dllExport);
+        // Generate declarations and definitions for all types except structs, classes, exceptions and proxies.
+        TypesVisitor typesVisitor(H);
         p->visit(&typesVisitor, false);
 
-        InterfaceVisitor interfaceVisitor(H, C, _dllExport);
-        p->visit(&interfaceVisitor, false);
+        // We need to fully declare proxy types before they can be used as struct and class fields.
+        ProxyVisitor proxyVisitor(H, C, _dllExport);
+        p->visit(&proxyVisitor, false);
+
+        // It's ok to use forward-declared classes as fields.
+        StructVisitor structVisitor(H);
+        p->visit(&structVisitor, false);
```

Comment:
Outside the context of this PR, this comment doesn't really make any sense on StructVisitor.

Reply:
I don't understand your comment.

Reply:
If you look at this code, which is for a struct visitor, and read the comment, it doesn't give enough information about why it's relevant for a struct visitor.

```
// It's okay to generate code for structs which maybe contain forward-declared classes.
```